src/ray/core_worker/core_worker.cc

```diff
@@ -271,16 +274,19 @@ CoreWorker::CoreWorker(const CoreWorkerOptions &options, const WorkerID &worker_
             new rpc::CoreWorkerClient(addr, *client_call_manager_));
       });
 
-  // Register a callback to monitor removed nodes.
+  // Register a callback to monitor add/removed nodes.
   // Note we capture a shared ownership of reference_counter_
   // here to avoid destruction order fiasco between gcs_client and reference_counter_.
-  auto on_node_change = [reference_counter = this->reference_counter_](
+  auto on_node_change = [this, reference_counter = this->reference_counter_](
                             const NodeID &node_id, const rpc::GcsNodeInfo &data) {
     if (data.state() == rpc::GcsNodeInfo::DEAD) {
       RAY_LOG(INFO) << "Node failure from " << node_id
                     << ". All objects pinned on that node will be lost if object "
                        "reconstruction is not enabled.";
       reference_counter->ResetObjectsOnRemovedNode(node_id);
+      num_alive_nodes_--;
+    } else {
+      num_alive_nodes_++;
```

Comment:
should we dump this and / or the value from get_max_pending_lease_requests_per_scheduling_category  to the state dump or somewhere so we can debug if this value is set correctly based on gcs events ? for example this value should default to 1 for a single node cluster?